osutil/epoll/epoll.go

```diff
@@ -128,11 +170,17 @@ func (e *Epoll) Deregister(fd int) error {
 //
 // Please refer to epoll_ctl(2) and EPOLL_CTL_MOD for details.
 func (e *Epoll) Modify(fd int, mask Readiness) error {
+	if e.IsClosed() {
+		return ErrEpollClosed
+	}
 	err := unix.EpollCtl(e.fd, unix.EPOLL_CTL_MOD, fd, &unix.EpollEvent{
 		Events: uint32(mask),
 		Fd:     int32(fd),
 	})
 	runtime.KeepAlive(e)
+	if e.IsClosed() {
```

Comment:
same, also with this I don't think we need the KeepAlive anymore as e is used after the syscall now